packages/experimental/navigation/hooks/__tests__/useNavigationTracker.test.tsx

```diff
@@ -0,0 +1,67 @@
+import {ForwardedRef} from 'react';
+import {renderHook} from '@testing-library/react-native';
+
+import useNavigationTracker, {type NavRef} from '../useNavigationTracker';
+
+jest.mock('../../otel/hooks/useSpan', () => () => ({
+  current: null,
+}));
+
+const mockSpanSetAttribute = jest.fn();
+const mockSpanEnd = jest.fn();
+
+const mockAddListener = jest.fn();
+const mockStartSpan = jest.fn(() => ({
+  end: mockSpanEnd,
+  setAttribute: mockSpanSetAttribute,
+  spanContext: jest.fn(),
+  setAttributes: jest.fn(),
+  addEvent: jest.fn(),
+  setStatus: jest.fn(),
+  updateName: jest.fn(),
+  isRecording: jest.fn(),
+  recordException: jest.fn(),
+}));
+const mockStartActiveSpan = jest.fn();
+const mockGetCurrentRoute = jest.fn();
+
+const mockTracerRef = {
+  current: {
+    startSpan: mockStartSpan,
+    startActiveSpan: mockStartActiveSpan,
+  },
+};
+
+const mockNavigationRef = {
+  current: {
+    // keep mocking functions when needed
+    addListener: mockAddListener,
+    getCurrentRoute: mockGetCurrentRoute,
+  },
+} as unknown as ForwardedRef<NavRef>;
+
+describe('useNavigationTracker.ts', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should work as expected when the application implements expo-router', () => {
+    renderHook(() => useNavigationTracker(mockNavigationRef, mockTracerRef));
+    expect(mockAddListener).toHaveBeenCalledWith('state', expect.any(Function));
+
+    const mockStateListenerCall = mockAddListener.mock.calls[0][1];
+
+    mockGetCurrentRoute.mockReturnValue({name: 'first-view-test'});
+    mockStateListenerCall();
+    expect(mockSpanEnd).not.toHaveBeenCalled();
+    // starts the initial span
+    expect(mockStartSpan).toHaveBeenCalledTimes(1);
+
+    mockGetCurrentRoute.mockReturnValue({name: 'second-view-test'});
```

Comment:
I'd be curious to take another look at these tests with less pieces mocked, e.g. I think we could use an actual tracerProvider and just need to supply it with a exporter that is mocked and in that way assert on the full body of the spans that would have been written out